ansible-sidecar/app/server.py

```
@@ -0,0 +1,181 @@
+        self.send_header("Connection", "close")
+        self.end_headers()
+
+        stop = threading.Event()
+
+        def heartbeat() -> None:
+            while not stop.wait(HEARTBEAT_SECONDS):
+                try:
+                    self.wfile.write(b": ping\n\n")
+                    self.wfile.flush()
+                except OSError:
+                    return
```

Ruff (0.16.0)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)